### PR TITLE
app-admin/keepassxc: Add X USE flag

### PR DESCRIPTION
**Pull request by mrusme**
Signed-off-by: マリウス <marius@xn--gckvb8fzb.com>

### CODE REVIEW

File: app-admin/keepassxc/metadata.xml [hunk 1]
```diff
@@ -15,6 +15,7 @@
 		<flag name="keeshare">Enable KeeShare sharing integration</flag>
 		<flag name="network">Enable network support</flag>
 		<flag name="yubikey">Enable database unlocking via YubiKey.</flag>
+		<flag name="X">Add support for X11</flag>
```
Comment (xgqt) [inline app-admin/keepassxc/metadata.xml:18]: not necessary, "X" is a global USE flag

File: app-admin/keepassxc/keepassxc-9999.ebuild [hunk 2]
```diff
@@ -25,7 +25,7 @@ fi
 
 LICENSE="LGPL-2.1 GPL-2 GPL-3"
 SLOT="0"
-IUSE="autotype browser doc keeshare +network test yubikey"
+IUSE="autotype browser doc keeshare +network test yubikey X"
```
Comment (xgqt) [inline app-admin/keepassxc/keepassxc-9999.ebuild:28]: "X" alphabetically is before "a" ;^)

**Comment by xgqt** [comment]
revision bump is not needed in this case

**Comment by mrusme** [comment]
@xgqt thank you, please check again!

File: app-admin/keepassxc/keepassxc-9999.ebuild [hunk 3]
```diff
@@ -39,7 +39,7 @@ RDEPEND="
 	dev-qt/qtnetwork:5
 	dev-qt/qtsvg:5
 	dev-qt/qtwidgets:5
-	dev-qt/qtx11extras:5
+	X? ( dev-qt/qtx11extras:5 )
```
Comment (xgqt) [inline app-admin/keepassxc/keepassxc-9999.ebuild:42]: oh, also the conditional dependencies should be below normal ones,
so this should be before "autotype?"

Comment (mrusme) [reply app-admin/keepassxc/keepassxc-9999.ebuild]: Sorry, corrected!

**Comment by xgqt** [comment]
While testing this I get the warring:

```
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    WITH_XC_X11
```

Looks like keepassxc does not support disabling X...?

**Comment by xgqt** [comment]
I see only live version supports it, so please do not add it to `2.7.1`.

**Comment by xgqt** [comment]
Also, for live `autotoype` gets disabled when `X11` is not enabled, so we need USE constraints
https://devmanual.gentoo.org/ebuild-writing/variables/index.html#required_use

**Comment by xgqt** [comment]
https://github.com/keepassxreboot/keepassxc/blob/develop/CMakeLists.txt#L115

**Comment by mrusme** [comment]
@xgqt good catch, corrected.

File: app-admin/keepassxc/keepassxc-9999.ebuild [hunk 4]
```diff
@@ -25,9 +25,10 @@ fi
 
 LICENSE="LGPL-2.1 GPL-2 GPL-3"
 SLOT="0"
-IUSE="autotype browser doc keeshare +network test yubikey"
+IUSE="X autotype browser doc keeshare +network test yubikey"
 
 RESTRICT="!test? ( test )"
+REQUIRED_USE="X? ( !autotype )"
```
Comment (xgqt) [inline app-admin/keepassxc/keepassxc-9999.ebuild:31]: I think there was a little misunderstanding, `X` requires `autotype`, so `X? ( autotype )`.

Comment (xgqt) [reply app-admin/keepassxc/keepassxc-9999.ebuild:31]: sorry, `autotype? ( X )`

**Comment by xgqt** [comment]
REQUIRED_USE should be other way around: `autotype? ( X )`
https://github.com/gentoo/gentoo/pull/27462#discussion_r979684716

**Comment by xgqt** [comment]
Thank You!

**Pull request closed without merging**